lighthouse-core/gather/gatherers/js-usage.js

```diff
@@ -72,73 +72,35 @@ class JsUsage extends FRGatherer {
     session.off('Debugger.scriptParsed', this.onScriptParsed);
   }
 
-  /**
-   * Usages alone do not always generate an exhaustive list of scripts in timespan and snapshot.
-   * For audits which use this for url/scriptId mappings, we can include an empty usage object.
-   *
-   * @param {Record<string, Array<LH.Crdp.Profiler.ScriptCoverage>>} usageByUrl
-   */
-  _addMissingScriptIds(usageByUrl) {
-    for (const scriptParsedEvent of this._scriptParsedEvents) {
-      const url = scriptParsedEvent.embedderName;
-      if (!url) continue;
-
-      const scripts = usageByUrl[url] || [];
-      if (!scripts.find(s => s.scriptId === scriptParsedEvent.scriptId)) {
-        scripts.push({
-          url,
-          scriptId: scriptParsedEvent.scriptId,
-          functions: [],
-        });
-      }
-      usageByUrl[url] = scripts;
-    }
-  }
-
   /**
    * @param {LH.Gatherer.FRTransitionalContext} context
    * @return {Promise<LH.Artifacts['JsUsage']>}
    */
   async getArtifact(context) {
-    /** @type {Record<string, Array<LH.Crdp.Profiler.ScriptCoverage>>} */
-    const usageByUrl = {};
+    /** @type {Record<string, LH.Crdp.Profiler.ScriptCoverage>} */
+    const usageByScriptId = {};
 
+    // TODO: remove?
```

Comment:
Oh, the `Force `Debugger.scriptParsed` events for url to scriptId mappings in snapshot mode.` bit, now that this gatherer isn't used to map script ids to url can the condition be removed? the TODO here was basically ask you :P

Reply:
I think snapshot support might still be useful in `JsUsage` to get treemap support in snapshot.

Reply:
On further investigation, we could probably just return `{}` in `JsUsage` for snapshot and it would work the same for the treemap. I think we can remove this.

Reply:
will remove in a follow up.